### PR TITLE
SHOW SHARDS

### LINKED ISSUE #3562: Should be able to show shards
For users to see what shards they have and where they're located, we need a command to show that information. I propose:

```
SHOW SHARDS
```

The result set should include the columns: `id, start_time, end_time, server_ids` Where `server_ids` is a string of comma separated server ids that have a copy of the shard.

Results should be ordered by shard ID.

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This pull request adds the ability to list all shards in the cluster and return their id, start time, end time, expiry time, and owner ids. Shards are grouped by database.

Fixes #3562

### CODE REVIEW

File: meta/statement_executor.go [hunk 1]
```diff
@@ -281,6 +286,45 @@ func (e *StatementExecutor) executeShowContinuousQueriesStatement(stmt *influxql
...
 }
 
+func (e *StatementExecutor) executeShowShardsStatement(stmt *influxql.ShowShardsStatement) *influxql.Result {
+	dis, err := e.Store.Databases()
+	if err != nil {
+		return &influxql.Result{Err: err}
+	}
+
+	rows := []*influxql.Row{}
+	for _, di := range dis {
```
Comment (otoolep) [inline meta/statement_executor.go]: What do you think about a `Shards()` method in `meta/store.go`?

File: meta/statement_executor.go [hunk 2]
```diff
@@ -281,6 +286,45 @@ func (e *StatementExecutor) executeShowContinuousQueriesStatement(stmt *influxql
...
+		row := &influxql.Row{Columns: []string{"id", "start_time", "end_time", "expiry_time", "owners"}, Name: di.Name}
+		for _, rpi := range di.RetentionPolicies {
+			for _, sgi := range rpi.ShardGroups {
+				for _, si := range sgi.Shards {
```
Comment (otoolep) [inline meta/statement_executor.go]: You'll only want to show shards from groups that haven't been tombstoned. Checked `Deleted()` on shard groups to see what I mean.

File: meta/statement_executor_test.go [hunk 3]
```diff
@@ -765,6 +765,53 @@ func TestStatementExecutor_ExecuteStatement_Unsupported(t *testing.T) {
...
+			},
+		}, nil
+	}
+
+	if res := e.ExecuteStatement(influxql.MustParseStatement(`SHOW SHARDS`)); res.Err != nil {
+		t.Fatal(res.Err)
+	} else if !reflect.DeepEqual(res.Series, influxql.Rows{
+		{
+			Name:    "foo",
+			Columns: []string{"id", "start_time", "end_time", "expiry_time", "owners"},
```
Comment (otoolep) [inline meta/statement_executor_test.go]: I think expiry time might confuse people. Thoughts? Is there any point in showing it?

**Comment by otoolep** [comment]
The flip side of this would be too show "tombstone" status, so we get full shard details in this diagnostic dump.

**Comment by otoolep** [comment]
CHANGELOG also needs updating.

File: meta/statement_executor_test.go (hunk 3 above)
Comment (beckettsean) [inline meta/statement_executor_test.go]: @otoolep wouldn't `expiry_time` help users understand why shards are sticking around despite a retention policy duration change? We agreed we shouldn't reset the `expiry_time` for those shards, and having output that confims that makes it much easier to describe and diagnose.

Comment (otoolep) [inline meta/statement_executor_test.go]: Yeah, on seconds thoughts more information is better.

File: meta/statement_executor.go (hunk 1 above)
Comment (dgnorton) [inline meta/statement_executor.go]: I've got one on the `dmq-show-measurements2` branch, which is ready to merge.  @benbjohnson  can merge this as-is and I can convert this to use the one in `meta` package while doing other DMQ work.

**Comment by pauldix** [comment]
+1

**Comment by benbjohnson** [comment]
I swear I thought I merged this PR. @otoolep I'm going to keep tombstoned shards in for now just to get more info. If it's too much then maybe we can add a `SHOW ALL SHARDS` statement that differentiates between active and all shards.

**Pull request merged**